### PR TITLE
Pushdown simplest sorting for `find` command

### LINKED ISSUE #2438: Pushdown simplest sorting with `ORDER BY`
### What should be done?

Currently, we rely on our Go code to sort documents the same way MongoDB does because PostgreSQL and other backends don't sort them the same way. But that requires us to fetch all documents before we can sort them, and that kills performance. Let's start investigating what we could do about that.

Let's add a new flag `--test-enable-sorting-pushdown` that would pushdown `ORDER BY field ASC/DESC` query part if:
* the command is either `find` or `aggregate`;
* `find`'s command `projection` parameter is empty;
* there is only one field name, and it does not use a dot notation.

If pushdown was performed, we should skip sorting with the Go code.

That flag should also be added to integration tests. No new automated tests or CI configurations are necessary.

### Where?

`pgdb`.

### Definition of Done

- PostgreSQL handler updated;
- spot refactorings done;
- tested manually.

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #2438.

The pushdown for the `find` command. The `aggregation` will come soon in other PR, just need to clarify some concerns.

The PR also changes explain command to use internal data to specify if pushdown occured, rather than using Index cond/filter fields from postgres. 

## Readiness checklist

* [x] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by w84thesun — DISMISSED** [review]
LGTM!

**Review by rumyantseva — DISMISSED** [review]
The idea and implementation looks good to me. To unblock you for aggregation implementation, I approve this PR.

To add in the second PR (apart from aggregation):
- a (unit) test for `iter, queryRes, err := QueryDocuments(ctxGet, tx, qp)` that `queryRes` is returned correctly
- check code cov if branches for sort pushdown are covered

**Comment by rumyantseva** [comment]
(Please fix the failing tests)

**Comment by noisersup** [comment]
It seems that integration is not panicing right now, I've also added unit tests.

File: internal/handlers/pg/msg_explain.go [hunk 1]
```diff
@@ -104,18 +104,18 @@ func (h *Handler) MsgExplain(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
...
 		return err
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	// if the plan returned filter info or index info, it means that pushdown had been done
-	pushdown := queryPlanner.HasByPath(types.NewStaticPath("Plan", "Filter")) ||
-		queryPlanner.HasByPath(types.NewStaticPath("Plan", "Index Cond"))
+	pushdown := results.FilterPushdown
```
Comment (rumyantseva) [inline internal/handlers/pg/msg_explain.go:118]: As you already have all the necessary things for `SortPushdown`, let's include it too?

**Review by rumyantseva — APPROVED** [review]
Tested locally, looks good to me.
One tiny suggestion - as it costs nothing to include the info about sort pushdown to explain, let's add it.

**Review by w84thesun — DISMISSED** [review]

File: internal/handlers/pg/msg_explain.go (hunk 1 above)
Comment (noisersup) [reply internal/handlers/pg/msg_explain.go]: done

**Review by rumyantseva — APPROVED** [review]
Great!

I tried it with and without the flag, works as expected! 


```
test> db.foo.find().sort("hello").explain()
{
  queryPlanner: {
    Plan: {
      'Node Type': 'Sort',
      'Parallel Aware': false,
      'Startup Cost': 97.78,
      'Plan Width': 64,
      Plans: [
        {
          'Parent Relationship': 'Outer',
          'Parallel Aware': false,
          'Async Capable': false,
          Schema: 'test',
          Alias: 'foo_a9f37ed7',
          Output: [ '_jsonb', "(_jsonb -> 'hello'::text)" ],
          'Node Type': 'Seq Scan',
          'Startup Cost': 0,
          'Total Cost': 27,
          'Plan Rows': 1360,
          'Plan Width': 64,
          'Relation Name': 'foo_a9f37ed7'
        }
      ],
      'Async Capable': false,
      'Total Cost': 101.18,
      'Plan Rows': 1360,
      Output: [ '_jsonb', "((_jsonb -> 'hello'::text))" ],
      'Sort Key': [ "((foo_a9f37ed7._jsonb -> 'hello'::text))" ]
    }
  },
  explainVersion: '1',
  command: { find: 'foo', filter: {}, sort: { hello: 1 }, '$db': 'test' },
  pushdown: false,
  sortingPushdown: true,
  serverInfo: {
    host: 'ferretdb',
    version: '6.0.42',
    gitVersion: 'bc504ad73fbe30deb7096cc82d76f6ba8370cbaa',
    ferretdbVersion: 'v1.0.0-95-gbc504ad7-dirty'
  },
  ok: 1
}
```

**Review by w84thesun — APPROVED** [review]

**Review by ptrfarkas — APPROVED** [review]

**Pull request merged**